LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -94,14 +97,26 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
             // Contains the Cloud to message we need to send
             IReceivedLoRaCloudToDeviceMessage cloudToDeviceMessage = null;
 
+            var skipDownstreamToAvoidCollisions = false;
+            var concentratorDeduplicationResult = this.concentratorDeduplication.CheckDuplicateData(request, loRaDevice);
+            if (concentratorDeduplicationResult is ConcentratorDeduplicationResult.Duplicate)
+            {
+                return new LoRaDeviceRequestProcessResult(loRaDevice, request, LoRaDeviceRequestFailedReason.DeduplicationDrop);
+            }
+            else if (concentratorDeduplicationResult is ConcentratorDeduplicationResult.SoftDuplicateDueToDeduplicationStrategy)
+            {
+                // Request is allowed upstream but confirmation is skipped to avoid collisions on the air.
```

Comment:
nit: 
```suggestion
                // Request is allowed upstream but confirmation is skipped to avoid sending the answer to the device multiple times and potentially cause collisions on the air.
```